improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -395,8 +395,11 @@ def process(
                 no_of_percentiles = len(
                     forecast_at_percentiles.coord(percentile_coord).points
                 )
+            minimum_percentile = percentile_coord.points.min()
             percentiles = choose_set_of_percentiles(
-                no_of_percentiles, sampling=sampling
+                no_of_percentiles,
+                sampling=sampling,
+                minimum_percentile=minimum_percentile,
```

Comment:
The disadvantage of this approach using the `minimum_percentile` from the inputs to constrain the percentiles output after resampling, is that this approach can no longer be considered as sampling quantiles. As the resampling is therefore no longer producing equally spaced quantiles, these percentiles can no longer be robustly converted to realizations. 

In the updates to the acceptance test data as part of this PR, this issue has been circumvented by using percentiles of 1 and 99, so that the `minimum_percentile` value isn't relevant, I think. Therefore I don't think that the `minimum_percentile` value is used by the acceptance tests. However, the changes in this PR are implicitly dependent upon the `minimum_percentile` being less than `1 / float(1 + no_of_percentiles)` otherwise equally-spaced quantiles are not created (and any subsequent rebadging of percentiles to realizations would result in an error).

I think that the main aim here is to avoid the use of the ECC bounds, so one option would be to make the use of the ECC bounds optional. I've tried this in principle, and the result is that the output percentiles outside of the range given by the input percentiles are effectively filled using nearest neighbour interpolation (i.e. assuming constant values). Would that be suitable? There are other options, so it would be worth a chat.

Reply:
As discussed, I agree that suppressing the use of ECC bounds would be easier for a hot-fix. I will add an option to prevent ECC bounds being used (as a separate PR) and test this in the suite. It will be much less impactful as a change than adding extra percentiles everywhere.

Gavin's point that this PR allows quantile sampling to output non-quantile data is important and this behaviour should not be permitted.